src/main/java/seedu/address/logic/commands/MemberCommand.java

```diff
@@ -0,0 +1,43 @@
+package seedu.address.logic.commands;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import seedu.address.logic.CommandHistory;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.exceptions.InvalidCommandModeException;
+import seedu.address.model.AppMode;
+import seedu.address.model.Model;
+
+/**
+ * Represents a command with hidden internal logic and the ability to be executed.
+ */
+public abstract class MemberCommand extends Command {
+
+    private ArrayList<AppMode.Modes> modeList = new ArrayList <AppMode.Modes> (
```

Comment:
This is violating abstraction. Command should not know which command mode it should be in AppMode.